Tools/CISupport/ews-build/steps.py

```diff
@@ -405,49 +405,76 @@ def evaluateCommand(self, cmd):
         return rc
 
 
-class ApplyPatch(shell.ShellCommand, CompositeStepMixin):
-    name = 'apply-patch'
-    description = ['applying-patch']
-    descriptionDone = ['Applied patch']
+class ApplyChange(steps.ShellSequence, CompositeStepMixin):
+    name = 'apply-change'
+    description = ['applying-change']
+    descriptionDone = ['Applied change']
     haltOnFailure = False
-    command = ['perl', 'Tools/Scripts/svn-apply', '--force', '.buildbot-diff']
 
     def __init__(self, **kwargs):
-        super(ApplyPatch, self).__init__(timeout=10 * 60, logEnviron=False, **kwargs)
+        super(ApplyChange, self).__init__(timeout=10 * 60, logEnviron=False, **kwargs)
 
     def _get_patch(self):
         sourcestamp = self.build.getSourceStamp(self.getProperty('codebase', ''))
         if not sourcestamp or not sourcestamp.patch:
             return None
         return sourcestamp.patch[1]
 
-    def start(self):
+    def run_bugzilla(self):
+        command = ['perl', 'Tools/Scripts/svn-apply', '--force', '.buildbot-diff']
+
         patch = self._get_patch()
         if not patch:
             # Forced build, don't have patch_id raw data on the request, need to fech it.
             patch_id = self.getProperty('patch_id', '')
-            self.command = ['/bin/sh', '-c', 'curl -L "https://bugs.webkit.org/attachment.cgi?id={}" -o .buildbot-diff && {}'.format(patch_id, ' '.join(self.command))]
-            shell.ShellCommand.start(self)
-            return None
+            self.commands.append(util.ShellArg(command=['curl', '-L', 'https://bugs.webkit.org/attachment.cgi?id={}'.format(patch_id), '-o', '.buildbot-diff'], logname='stdio'))
+            self.commands.append(util.ShellArg(command=command, logname='stdio'))
+            return steps.ShellSequence.run(self)
 
         patch_reviewer_name = self.getProperty('patch_reviewer_full_name', '')
         if patch_reviewer_name:
-            self.command.extend(['--reviewer', patch_reviewer_name])
+            command.extend(['--reviewer', patch_reviewer_name])
+        self.commands.append(util.ShellArg(command=command, logname='stdio'))
+
         d = self.downloadFileContentToWorker('.buildbot-diff', patch)
-        d.addCallback(lambda res: shell.ShellCommand.start(self))
+        d.addCallback(lambda res: steps.ShellSequence.run(self))
+        return d
+
+    def run_github(self):
+        remote = self.getProperty('github.head.repo.full_name', 'origin').split('/')[0]
+        project = self.getProperty('github.head.repo.full_name', self.getProperty('project'))
+        branch = self.getProperty('github.head.ref', 'main')
+
+        for command in [
+            ['/bin/sh', '-c', 'git remote add {} {}{}.git & true'.format(remote, GITHUB_URL, project)],
```

Comment:
`& true` seems to be for debugging

Reply:
This actually isn't for debugging, it's because `git` doesn't let me say "add or update" in a single command. If the checkout is already tracking to specified remote, this command will fail. Which is a failure we're interested in ignoring. Open to other ways of expressing that, this was the simplest one I could come up with.